extensions/amp-analytics/0.1/visibility-manager.js

```diff
@@ -230,11 +231,69 @@ export class VisibilityManager {
    */
   listenElement(
       element, spec, readyPromise, createReportPromiseFunc, callback) {
-    const model = new VisibilityModel(
-        spec,
-        this.getElementVisibility.bind(this, element));
-    return this.listen_(
-        model, spec, readyPromise, createReportPromiseFunc, callback, element);
+    const calcVisibility = this.getElementVisibility.bind(this, element);
+    return this.createModelAndListen_(calcVisibility, spec, readyPromise,
+        createReportPromiseFunc, callback, element);
+  }
+
+  /**
+   * Create visibilityModel and listen to visible events.
+   * @param {function():number} calcVisibility
+   * @param {!Object<string, *>} spec
+   * @param {?Promise} readyPromise
+   * @param {?function():!Promise} createReportPromiseFunc
+   * @param {function(!Object<string, *)} callback
+   * @param {!Element=} opt_element
+   * @return {!UnlistenDef}
+   */
+  createModelAndListen_(calcVisibility, spec,
+      readyPromise, createReportPromiseFunc, callback, opt_element) {
+    if (spec['visiblePercentageThresholds'] &&
+        spec['visiblePercentageMin'] == undefined &&
+        spec['visiblePercentageMax'] == undefined) {
+      const unlisteners = [];
+      //const ranges = tryParseJson(spec['visiblePercentageThresholds']);
+      const ranges = spec['visiblePercentageThresholds'];
+      if (!ranges || !isArray(ranges)) {
+        user().error(TAG, 'invalid visiblePercentageThresholds');
+      }
+      console.log('ranges is ', ranges);
```

Comment:
Nit: Remove console.log() calls? Also, may want to return after user().error() above.

Reply:
good point. done